test/phoenix/digester_test.exs

```diff
@@ -0,0 +1,50 @@
+defmodule Phoenix.DigesterTest do
+  use ExUnit.Case, async: true
+
+  test "fails when the given paths are invalid" do
+    assert {:error, :invalid_path} = Phoenix.Digester.compile("../some invalid?/path", "/other_invalid ?? /path")
+  end
+
+  test "digests and compress files" do
+    output_path = Path.join("tmp", "compiled")
+    input_path = "priv/static"
+    File.mkdir(output_path)
+
+    assert :ok = Phoenix.Digester.compile(input_path, output_path)
+
+    [input_file, _]= assets_files(input_path)
+    output_files = assets_files(output_path)
+
+    name = Path.rootname(input_file)
+    assert Enum.member?(output_files, input_file)
+    assert Enum.member?(output_files, "manifest.json")
+    assert Enum.member?(output_files, input_file <> ".gz")
+    assert Enum.any?(output_files, &(String.match?(&1, ~r/(#{name}-[a-fA-F\d]{32})/)))
+  end
+
+  test "doesn't duplicate files when digesting and compressing twice" do
+    path = Path.join("tmp", "static_files")
+    input_file = Path.join(path, "file.js")
+    File.mkdir_p!(path)
+    File.write!(input_file, "console.log('test');")
+
+    assert :ok = Phoenix.Digester.compile(path, path)
+    assert :ok = Phoenix.Digester.compile(path, path)
```

Comment:
Its digesting twice for the same path. It is now checking if it's duplicating the `.gz` and the digest.